src/Filesystem/File.php

```diff
@@ -556,11 +556,12 @@ public function mime()
         }
         if (function_exists('finfo_open')) {
             $finfo = finfo_open(FILEINFO_MIME);
-            $finfo = finfo_file($finfo, $this->pwd());
             if (!$finfo) {
                 return false;
             }
-            list($type) = explode(';', $finfo);
+            $mime = finfo_file($finfo, $this->pwd());
```

Comment:
i dont see any throws in:
https://github.com/php/php-src/blob/PHP-5.4.1/ext/fileinfo/fileinfo.c

so looks like all that is needed is
```PHP
$finfo = new finfo(FILEINFO_MIME);
if (!$finfo ) {
    return false;
}
$type = $finfo->file($this->pwd());
if (!$type) {
    return false;
}
list($type) = explode(';', $type);
return $type;
```
as i understand it will autoclose on destroy